src/libraries/System.Text.Json/gen/Helpers/RoslynExtensions.cs

```diff
@@ -209,28 +209,6 @@ public static bool IsNullableValueType(this ITypeSymbol type, [NotNullWhen(true)
             return false;
         }
 
-        public static ITypeSymbol[] GetAllTypeArgumentsInScope(this INamedTypeSymbol type)
```

Comment:
Taking a closer look at this algorithm, it seems that for a shape like
```C#
public class Parent<T>
{
    public class Child<U> {}
}
```
given the symbol for `Parent<int>.Child<string>` we'll be generating the identifier `Child_int_string`, whereas one might have reasonably expected `Parent_int_Child_string`. 

We obviously can't change that in STJ for fear of source breaking changes, but is it really something we want to introduce in the config binder or even proliferate it to future generator by extracting that logic to the common folder?

Reply:
Aren't these identifiers hidden from the user, so we could change this algorithm without compat impact?

Reply:
Agreed. The APIs are `file` scoped.. so we can change them as needed.